elyra/pipeline/component_catalog.py

```diff
@@ -44,40 +48,144 @@
-
         while not self.stop_event.is_set():
-            timeout = NONBLOCKING_TIMEOUT if outstanding_threads else BLOCKING_TIMEOUT
             try:
-                # Get a task from the queue
-                catalog, operation = self._cache_task_queue.get(timeout=timeout)
-
+                # Get a task from the manifest queue
+                catalog, action = self._manifest_queue.get(timeout=NONBLOCKING_TIMEOUT)
             except Empty:
-                # No task exists in the queue, proceed to check for thread execution
+                # No task exists in the manifest queue, proceed to check for cache tasks
                 pass
-
             else:
-                # Create and start a thread for the task
-                updater_thread = CacheUpdateThread(self._component_cache, self._cache_task_queue, catalog, operation)
-                updater_thread.start()
+                # Make the required update to the manifest file
+                if os.path.isfile(self._manifest_filename):
+                    with open(self._manifest_filename, 'r') as f:
+                        manifest = json.loads(f.read())
+
+                write_required = False
+                if action == 'cache':
+                    # A 'cache' action means that updates to the component cache are required
+
+                    # Handle all pending actions in one batch
+                    for pending_action in manifest.get('actions'):
+                        catalog_name = list(pending_action.keys())[0]
+                        cache_action = list(pending_action.values())[0]
+
+                        write_required = True
+
+                        catalog_instance: ComponentCatalogMetadata = None
+                        if cache_action == 'delete':
+                            # Remove status stanza from manifest
+                            manifest['status'].pop(catalog_name, None)
+
+                            # Fabricate a metadata instance that only includes catalog name
+                            catalog_instance = ComponentCatalogMetadata(name=catalog_name)
+
+                        elif cache_action == 'modify':
+                            # Create or replace the status stanza to indicate status is current
+                            errors = manifest['status'].get(catalog_name, {}).get('errors', [])
+                            manifest['status'][catalog_name] = {
+                                'action': 'current',
+                                'errors': errors
+                            }
+
+                            # Fetch the catalog instance associated with this action
+                            catalog_instance = MetadataManager(
+                                schemaspace=ComponentCatalogs.COMPONENT_CATALOGS_SCHEMASPACE_ID
+                            ).get(name=catalog_name)
+
+                        # Add action to the cache queue
+                        self._cache_queue.put((catalog_instance, cache_action))
```

Comment:
What does it mean to place a `None` catalog_instance into the cache queue?  If this shouldn't happen perhaps the `elif` could be `else`.